log/handlers.ts

```diff
@@ -180,18 +178,30 @@ export class FileHandler extends WriterHandler {
   }
 
   log(msg: string) {
-    if (this._encoder.encode(msg).byteLength + 1 > this._buf.available()) {
+    const bytes = this._encoder.encode(msg + "\n");
+    if (bytes.byteLength > this._buf.byteLength - this._pointer) {
       this.flush();
     }
-    this._buf.writeSync(this._encoder.encode(msg + "\n"));
+    this._buf.set(bytes, this._pointer);
+    this._pointer += bytes.byteLength;
   }
 
   flush() {
-    if (this._buf?.buffered() > 0) {
-      this._buf.flush();
+    if (this._pointer > 0 && this._file) {
+      let written = 0;
+      while (written < this._pointer) {
+        written += this._file.writeSync(
```

Comment:
This is using the `WriterSync` interface, which is part of the deprecations in question. Are we able to use `file.writable` instead? Though, doing so would make `flush()` async.

Reply:
> Are you sure? I can see that [std/io/types.d.ts WriterSync](https://deno.land/std@0.210.0/io/types.d.ts?s=WriterSync) is deprecated but neither [Deno.FsFile](https://deno.land/api@v1.39.1?s=Deno.FsFile) nor [Deno.FsFile.prototype.writeSync](https://deno.land/api@v1.39.1?s=Deno.FsFile&p=prototype.writeSync) appear to be. Do the docs need to be updated to mark `Deno.FsFile.prototype.writeSync` as deprecated?

Yes, I'm sure. I've opened https://github.com/denoland/deno/pull/21701.

We'll get back to you on the other points.

Reply:
Actually, there are no plans to deprecate these `Deno.FsFile` methods. Let's continue with this PR 👍🏾